src/workerd/api/kv.c++

```diff
@@ -26,6 +26,8 @@ static void validateKeyName(kj::StringPtr name) {
   JSG_REQUIRE(name != "", TypeError, "Key name cannot be empty.");
   JSG_REQUIRE(name != ".", TypeError, "\".\" is not allowed as a key name.");
   JSG_REQUIRE(name != "..", TypeError, "\"..\" is not allowed as a key name.");
+  JSG_REQUIRE(name.size() <= 8 * 1024, TypeError, "UTF-8 encoded length of ", name.size(),
+      " exceeds limit of 8 KiB.");
```

Comment:
I'd go ahead and add a mention to "key name" here, since it's a little ambiguous otherwise.

And are we sure we want a higher limit here? It's a little confusing to get an error about `8 KiB`, then you try a 7 KiB key and get a totally different error, with a different key length. We haven't changed KV key length in forever (and I can't remember a single request about it), any reason not to just duplicate the same length/error here and worry about the tiny chance it changes later?

Reply:
Done.

Reply:
Any thoughts on the second bit?

Reply:
(possible that the error message expression may need to be `kj::str("UTF-8 encoded length of ", name.size(), " exceeds limit of 8 KiB.")` to ensure that it doesn't attempt to stringify `name.size() = ` into the message?)

Reply:
Nope. `JSG_REQUIRE` macros are different & don't do the normal `KJ` stringification.